### ISSUE TITLE
TypeError: Unknown file extension ".svelte"

### ISSUE THREAD

**Issue by dennismphil** [mhkeller/layercake#80]
In my svelte project, I am using TypeScript, vite and vitest.

When I included layercake, my unit tests are failing with the following error.

```
TypeError: Unknown file extension ".svelte" for /node_modules/.pnpm/layercake@6.1.0/node_modules/layercake/LayerCake.svelte
```

The failure message is seen only in unit tests. The chart works great in the app/browser.

All test passes if I remove the `type="module"` from layercakes' [package.json](https://github.com/mhkeller/layercake/blob/main/package.json#L19)

Any theories why my TypeScript project cannot read \*.svelte files in use in the imports of 3rd party modules(layercake)?

This is my `tsconfig.json`

```ts
{
  "extends": "@tsconfig/svelte/tsconfig.json",
  "compilerOptions": {
    "target": "esnext",
    "useDefineForClassFields": true,
    "module": "esnext",
    "resolveJsonModule": true,
    "baseUrl": ".",
    /**
     * Typecheck JS in `.svelte` and `.js` files by default.
     * Disable checkJs if you'd like to use dynamic types in JS.
     * Note that setting allowJs false does not prevent the use
     * of JS in `.svelte` files.
     */
    "allowJs": true,
    "checkJs": true,
    "strict": true,
    "types": ["vitest/globals", "svelte"],
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": [
    "src/**/*.d.ts",
    "src/**/*.ts",
    "src/**/*.js",
    "src/**/*.svelte",
    "test/**/*.ts"
  ],
  "references": [{ "path": "./tsconfig.node.json" }]
}
```

**Comment by dennismphil** [comment]
The solution here to add the following to my `vitest.config.ts`

```typescript
...
export default defineConfig({
  ...
  optimizeDeps: {
    exclude: [
      'layercake'
    ]
  },
  ...
});
```

**Comment by mhkeller** [comment]
Did I see you solved it by adding it to the exclude list under optimizeDeps? I saw that in an email notification but i don’t see the comment here

**Comment by dennismphil** [comment]
Thanks for the response.  Right, that did not seem to work. I jumped the gun and thought this would resolve the issue.

I see a related discussion here
https://github.com/sveltejs/kit/issues/3172

**Comment by mhkeller** [comment]
It could be an issue with how you’re running your tests and you need to add a plug-in to read svelte files. LayerCake is using the standard svelte kit setup and since it builds that tells me it it’s able to load svelte files in production and dev. The pure node environment in your tests may be the problem? But I find the current vite ecosystem a bit confusing and may not be the best person to debug your specific setup.

**Comment by dennismphil** [comment]
Yes, layercake works great in the browser and the application. This issue is seen only in the unit test which is quite confusing and i hit a wall now. I hope someone in the community who is knowledgeable can help

**Comment by dennismphil** [comment]
Repro of the issue  
  
https://stackblitz.com/edit/github-zrwj6k-gn8ljv?file=components/Hello.svelte

**Comment by dennismphil** [comment]
https://github.com/vitest-dev/vitest/issues/1395#issuecomment-1140670262

**Comment by mhkeller** [comment]
I’m glad you figured it out. What does this option do and should it be the default for example templates? Does it have any side effects when run in the browser?
